routes/adjustmentMethods/EToAdjustmentMethod.ts

```diff
@@ -0,0 +1,224 @@
+import * as SunCalc from "suncalc";
+import * as moment from "moment";
+import { AdjustmentMethod, AdjustmentMethodResponse, AdjustmentOptions } from "./AdjustmentMethod";
+import { GeoCoordinates, WateringData, WeatherProviderId } from "../../types";
+import { WeatherProvider } from "../weatherProviders/WeatherProvider";
+
+
+/**
+ * Calculates how much watering should be scaled based on weather and adjustment options by comparing the recent
+ * potential ETo to the baseline potential ETo that the watering program was designed for.
+ */
+async function calculateEToWateringScale(
+	adjustmentOptions: EToScalingAdjustmentOptions,
+	wateringData: WateringData | undefined,
```

Comment:
Observation: ETo is usually considered a loss that has already occurred, i.e., it's a backward look; the goal of irrigation is to replace what was lost.  If that's the intent here, then whether or not it is currently raining would not be relevant because the quantity of the current rain would be accounted for in the next ETo calculation.  Note, however, that this only works correctly if there's a way to accumulate ETo to match with the watering schedule.

Reply:
WateringData is included as a parameter because `weather.ts` will use it check watering restrictions, so it will be requested for every non-manual AdjustmentMethod. I can refactor this to clean up the code and reduce the number of requests needed by creating a `BaseWateringData` class that `EToData` and `ZimmermanData` (renamed from `WateringData` to clarify its purpose), and adding a `getWateringData` method to `AdjustmentMethod` that returns a sub-interface of `BaseWateringData`. This structure would look something like this:
```ts
interface BaseWateringData {
	weatherProvider: WeatherProviderId;
	precip: number;
}

interface EToData extends BaseWateringData {
	// All fields currently in EToData that aren't moved to BaseWateringData
}

interface ZimemrmanData extends BaseWateringData {
	// All fields currently in WateringData that aren't moved to BaseWateringData
}

interface AdjustmentMethod {
	// ...
	// Returns the data this AdjustmentMethod uses (a subclass of BaseWateringData)
	getWateringData( coordinates: GeoCoordinates ): Promise< BaseWateringData >;
}

interface AdjustmentMethodResponse {
	// ...
	wateringData: BaseWateringData;
}

// Although these do add an extra level of redirection, they help keep each AdjustmentMethod's logic isolated to its own file file (instead of handling all the logic in weather.ts)
interface EToAdjustmentMethod {
	// ...
	getWateringData( weatherProvider: WeatherProvider, coordinates: GeoCoordinates ): Promise< BaseWateringData > {
		return weatherProvider.getEToData( coordinates );
	}
}
interface ZimmermanAdjustmentMethod {
	// ...
	getWateringData( weatherProvider: WeatherProvider, coordinates: GeoCoordinates ): Promise< BaseWateringData > {
		// Renamed from "getWateringData".
		return weatherProvider.getZimmermanData( coordinates );
	}
}
```
What are your thoughts on this?

Reply:
Let me think on this one as it is at the heart of the matter.

Reply:
Resolving in favor of Slack discussion